File: week2/caesar/caesar.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <cs50.h>

int main(int argc, string argv[]) {
    if (argc < 2 || argc > 2) {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    int k = atoi(argv[1]) % 26;
    printf("plaintext:  ");
    string s = get_string();
    if (s != NULL) {
        printf("ciphertext: ");
        for (int i = 0, n = strlen(s); i < n; i++) {
            int ascii_s = (int)s[i];
            if (ascii_s > 64 && ascii_s < 91) {
                if ((ascii_s + k) > 90) {
                    printf("%c", ascii_s + k - 26);
                }
                else {
                    printf("%c", ascii_s + k);
                }
            }
            else if (ascii_s > 96 && ascii_s < 123) {
                if ((ascii_s + k) > 122) {
                    printf("%c", ascii_s + k - 26);
                }
                else {
                    printf("%c", ascii_s + k);
                }
            }
            else {
                printf("%c", s[i]);
            }
        }
        printf("\n");
    }
}

File: week8/caesar/caesar.py
import sys

def main():
    if len(sys.argv) != 2:
        print('Usage: ./caesar k')
        return 1
    k = int(sys.argv[1]) % 26

    plaintext = input('plaintext:  ')
    if plaintext == None:
        return 1

    print('ciphertext: ', end='');
    for s in plaintext:
        ascii_s = ord(s)
        if ascii_s > 64 and ascii_s < 91:
            if (ascii_s + k) > 90:
                print(chr(ascii_s + k - 26), end= '');
            else:
                print(chr(ascii_s + k), end= '');
        elif ascii_s > 96 and ascii_s < 123:
            if (ascii_s + k) > 122:
                print(chr(ascii_s + k - 26), end= '');
            else:
                print(chr(ascii_s + k), end = '');
        else:
            print(s, end = '');
    print('')
    return 0

if __name__ == '__main__':
    main()